attachment.go

```diff
@@ -0,0 +1,147 @@
+type Attachment struct {
+	ID string `json:"id,omitempty"`
+	Title string `json:"title,omitempty"`
+	Links struct {
+		Download string `json:"download,omitempty"`
+	} `json:"_links,omitempty"`
+}
+
+type AttachmentResults struct {
+	ResultPagination
+	Results []Attachment `json:"results"`
+}
+
+func (w *Wiki) getAttachmentByFilenameEndpoint(contentID string, filename string) (*url.URL, error) {
+	return url.ParseRequestURI(fmt.Sprintf("%v/content/%v/child/attachment?filename=%v", w.endPoint.String(), contentID, filename))
+}
+
+func (w *Wiki) createAttachmentEndpoint(contentID string) (*url.URL, error) {
+	return url.ParseRequestURI(fmt.Sprintf("%v/content/%v/child/attachment", w.endPoint.String(), contentID))
+}
+
+func (w *Wiki) updateAttachmentEndpoint(contentID string, attachmentID string) (*url.URL, error) {
+	return url.ParseRequestURI(fmt.Sprintf("%v/content/%v/child/attachment/%v/data", w.endPoint.String(), contentID, attachmentID))
+}
+
+func (w *Wiki) GetAttachment(contentID string, filename string) (*AttachmentResults, error) {
+	endpoint, err := w.getAttachmentByFilenameEndpoint(contentID, filename)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("GET", endpoint.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := w.sendRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var results AttachmentResults
+	err = json.Unmarshal(res, &results)
+	if err != nil {
+		return nil, err
+	}
+
+	return &results, nil
+}
+
+func (w *Wiki) GetAttachmentData(contentID string, filename string) ([]byte, error) {
+	result, err := w.GetAttachment(contentID, filename)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("GET", w.root.String() + result.Results[0].Links.Download, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := w.sendRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (w *Wiki) CreateAttachment(contentID string, path string) (*AttachmentResults, error) {
+	endpoint, err := w.createAttachmentEndpoint(contentID)
+	if err != nil {
+		return nil, err
+	}
+	return w.createOrUpdateAttachment(endpoint, path)
+}
+
+func (w *Wiki) UpdateAttachment(contentID string, path string, attachmentID string) (*AttachmentResults, error) {
+	endpoint, err := w.updateAttachmentEndpoint(contentID, attachmentID)
+	if err != nil {
+		return nil, err
+	}
+	return w.createOrUpdateAttachment(endpoint, path)
+}
+
+func (w *Wiki) createOrUpdateAttachment(endpoint *url.URL, path string) (*AttachmentResults, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("file", filepath.Base(path))
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = io.Copy(part, file)
```

Comment:
- [ ] Return the `io.Copy()` error

Reply:
Completed, but github won't let me check that checkbox.